src/modules/attendance/attendance.service.ts

```diff
@@ -0,0 +1,162 @@
+import { Injectable, BadRequestException, Inject } from '@nestjs/common';
+import { InjectRepository, InjectDataSource } from '@nestjs/typeorm';
+import { Repository, DataSource, EntityManager } from 'typeorm';
+import { AttendanceEntity } from './entities/attendance.entity';
+import type { IEventRegistrationService } from '../event/i-event-registration.service';
+import { AttendanceRegister } from './domain/attendace-register';
+import { Attendance } from './domain/attendance';
+import { AttendanceStatus } from './domain/attendance-status';
+import { EventService } from '../event/event.service';
+import { UserService } from '../user/user.service';
+import { EventStatus } from '../event/domain/event-status';
+import { AttendanceCancel } from './domain/attendance-cancel';
+
+@Injectable()
+export class AttendanceService {
+  constructor(
+    @InjectRepository(AttendanceEntity)
+    private attendanceRepository: Repository<AttendanceEntity>,
+
+    @InjectDataSource()
+    private dataSource: DataSource,
+
+    @Inject('IEventRegistrationService')
+    private eventRegistrationService: IEventRegistrationService,
+
+    private readonly eventService: EventService,
+
+    private readonly userService: UserService,
+
+  ) {}
+
+  async register(attendanceRegister: AttendanceRegister): Promise<Attendance> {
```

Comment:
Nếu cần transaction, nhưng cái này cũng không quá cần

```suggestion
  @Transactional()
  async register(attendanceRegister: AttendanceRegister): Promise<Attendance> {
```